metricbeat/mb/module/factory.go

```diff
@@ -43,28 +41,32 @@ func NewFactory(beatInfo beat.Info, options ...Option) *Factory {
 
 // Create creates a new metricbeat module runner reporting events to the passed pipeline.
 func (r *Factory) Create(p beat.Pipeline, c *common.Config, meta *common.MapStrPointer) (cfgfile.Runner, error) {
```

Comment:
For a future refactor, we should consider making this factory the only way to create modules. We have more places where we create module runners, wrappers and connectors.

Reply:
Are you referring to `metricbeat/beater/metricbeat.go` or a different place?

Reply:
Yes, where we instantiate the modules from "static" and "dynamic" configurations.